presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java

```diff
@@ -153,6 +157,16 @@ public StatsAndCosts getStatsAndCosts()
         return jsonRepresentation;
     }
 
+    // Serialize this plan fragment with the provided codec, caching the results
+    public synchronized byte[] toBytes(Codec<PlanFragment> codec)
```

Comment:
nit: after going through the code, it is clear the `Codec` should not change during the lifecycle of a JVM. Lets just add a comment that clarifies this here. Because technically if someone passes a different codec in the second call, we will be returning the incorrect result. Better if we can add some variant for the same.

Reply:
I'll add the reference to the first successfully used `Codec` and throw if it differs.